app/services/streams/twitch_access_token/get.rb

```diff
@@ -7,11 +7,11 @@ def self.call
       end
 
       def call
-        token, exp = Rails.cache.fetch(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY)
+        token, exp = RedisRailsCache.fetch(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY)
 
         if token.nil? || Time.zone.now >= exp
           token, exp = get_new_token
-          Rails.cache.write(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY, [token, exp])
+          RedisRailsCache.write(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY, [token, exp])
```

Comment:
I am SO SORRY this has taken a bit for us to get to! I was trying to get some additional color on the whole situation since I am still a bit new here 😃 One last thing before this is ready. This will not set the expiration like you would think according to my rough testing in a console. Let me know if that makes sense and if so we should update this method so the expiration is set correctly.
```
[38] pry(main)> RedisRailsCache.write("tok", [1, 1000])
=> "OK"
[40] pry(main)> RedisRailsCache.redis.ttl("tok")
=> 86385
[41] pry(main)> RedisRailsCache.redis.ttl("tok")
=> 86384
[42] pry(main)> RedisRailsCache.write("tok", 1, :expires_in => 1000)
=> "OK"
[43] pry(main)> RedisRailsCache.redis.ttl("tok")
=> 998
```

Reply:
Yes, I guess you are right, How should we update it ?

Reply:
Similar to @mstruve example:

```ruby
RedisRailsCache.write(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY, token, expires_in: exp)
```

instead of

```ruby
RedisRailsCache.write(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY, [token, exp])
```

currently it saves an array of two values instead of the token and setting its expiration date

Reply:
Thanks @rhymes :)

I have made apt changes @mstruve